source/application/AMPIC887DriverServer/AMPIC887AxisCollection.h

```diff
@@ -0,0 +1,77 @@
+#ifndef AMPIC887AXISCOLLECTION_H
+#define AMPIC887AXISCOLLECTION_H
+#include <QList>
+#include "AMGCS2.h"
+
+/*!
+  * A class representing a collection of axes controlled by a PI C887.11 Controller
+  */
+class AMPIC887AxisCollection : public QList<AMGCS2::Axis>
+{
+public:
+	// Enumerates the different states the collection can be initialized in.
+	enum InitializationState {
+		EmptyCollection,	// Collection is initialized with no contained axes
+		AllAxes,			// Collection is initialized with all axes
+		LinearAxes			// Collection is initialized with linear axes (X, Y, Z)
+	};
+
+	// Enumerates the different valid states which the colleciton can be in
+	enum ValidState {
+		Valid,					// Collection is valid
+		ContainsUnknownAxis,	// Collection contains an unknown axis
+		ContainsDuplicateAxes	// Collection contains duplicates
```

Comment:
True, that would probably be a good addition. I enumerated all the invalid states I'd yet encountered, but a general case would be nice.